src/app_service/service/community/service.nim

```diff
@@ -86,6 +86,52 @@ QtObject:
     result.allCommunities = initTable[string, CommunityDto]()
     result.myCommunityRequests = @[]
 
+  proc findChatById(id: string, chats: seq[ChatDto]): ChatDto =
+    for chat in chats:
+      if(chat.id == id):
+        return chat
+
+  proc findById(id: string, chats: seq[Chat]): int =
+    for chat in chats:
+      if(chat.id == id):
+        return 0
+    return -1
+
+  proc handleCommunityUpdates(self: Service, communities: seq[CommunityDto], updatedChats: seq[ChatDto]) =
+    let community = communities[0]
+    if(not self.joinedCommunities.hasKey(community.id)):
+      error "error: community doesn't exist"
+      return
+
+    let prev_community = self.joinedCommunities[community.id]
+
+    # channel was added
+    if(community.chats.len > prev_community.chats.len):
+      for chat in community.chats:
+        if findById(chat.id, prev_community.chats) == -1:
+          let updated_chat = findChatById(community.id&chat.id, updatedChats)
+          self.events.emit(SIGNAL_COMMUNITY_CHANNEL_CREATED, CommunityChatArgs(communityId: community.id, chat: toChat(updated_chat)))
```

Comment:
@Khushboo-dev-cpp I see what was your point with `toChat` conversion. But let's see it this way... since you already use `chat` in line 110 to iterate over `community.chats` which is by itself a `Chat`, doesn't it contain all necessary info you need? If it does just use it as a `chat` param of `CommunityChatArgs` instead of converting a `ChatDto` to `Chat`? If not then `toChat` proc you added, logically should not be part of `dto/community.nim` file, but move it here to the logic file of this module, make it private, and name it `mapChatDtoToChat` (but only if we really need it).

Reply:
So Sale, the thing is that the joinedCommunities chats correspond to the chats from the communities arg we received from the Signal. However those chats dont have the params filled in correctly, for ex: id, color etc. Hence I needed to access the receivedData.chats in order to update it on the UI side.
However I could just pass the ChatDto to the UI for all the case and avoid using chat so that they are all synced.

Reply:
> So Sale, the thing is that the joinedCommunities chats correspond to the chats from the communities arg we received from the Signal. However those chats dont have the params filled in correctly, for ex: id, color etc.

Exactly that was what I asked (and I supposed that it's like that), thanks. 

So in that case please do the following:
- revert changes you did in `dto/community.nim`
- add private proc `mapChatDtoToChat` to the module (which will be the same as `toChat` you added to `dto/community.nim`)
- use `mapChatDtoToChat` where you're currently using `toChat`

so it's a simple change (just move the thing to the right place).

Reply:
done